receiver/apachereceiver/internal/configgen/component_config_gen.go

```diff
@@ -0,0 +1,27 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/configschema"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachereceiver"
+)
+
+//go:generate go run ./component_config_gen.go
+
+func main() {
```

Comment:
Oh, we bring the file generation to the application runtime... I'm not sure it's a good approach to have documentation generation affect the application runtime. It might delay the collector start or break the collector on some platforms. We don't test mdatagen extensively on all platforms. I believe all the file generation should stay in specific commands or make targets, not the application itself.

I see why you are doing it, and I don't have any similar alternative suggestion, unfortunately.

If we cannot generate the files as part of `make generate` to remove the dependency on all modules from one place, I think we should avoid going through this step and go straight with metadata -> (config structs + docs) generation as was suggested in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27003.

Reply:
Sorry for not bringing this earlier

Reply:
Actually, nevermind what I wrote. I confused it with `init()` statement, lol. My head is blurry this morning